apps/api-extractor/src/generators/ApiModelGenerator.ts

```diff
@@ -330,10 +330,12 @@ export class ApiModelGenerator {
       const apiItemMetadata: ApiItemMetadata = this._collector.fetchApiItemMetadata(astDeclaration);
```

Comment:
> * **Did not** update any api-documenter packages yet - unsure how we'd want this information to be surfaced - open to suggestions!

This design question seems important, since the primary application of `api-extractor-model` is for generating website docs. Let's try to answer in this PR.  @zelliott I'd also be interested to hear how you plan to display it with your internal documentation tool.

Reply:
How about we just stick it in the "modifiers" column like this:

![image](https://user-images.githubusercontent.com/4673363/166847628-bc52e84f-a184-4f20-ab75-3295754f8e4d.png)

Reply:
> @zelliott I'd also be interested to hear how you plan to display it with your internal documentation tool.

Probably as a pill somewhere next to the property name, just as you've mocked up in the screenshot above. We might also add a toggle to show/hide properties of different visibility (e.g. show just `public`, or show `public` + `protected`).

Speaking of which, your screenshot above looks great to me. I'll go ahead and get this added, and request feedback once it's ready.

Reply:
@octogonz Having trouble figuring out how to get the modifier code spans to appear on separate lines. I thought placing them within distinct `DocParagraph`s would do the trick, but no luck. I pushed what I have so far - but thought I would solicit your help in parallel to my own investigation.

Reply:
This is a bug actually. I pushed a proposed fix -- take a look at https://github.com/microsoft/rushstack/pull/3387/commits/b6ad1388d607ca50f2141997bd24b357aea9e324

Reply:
Ahh cool, nice catch 👍

Reply:
It is a different bug:

https://github.com/microsoft/rushstack/blob/c7aef037ee57dbb0892b2882c9d8404cc51ce0fc/apps/api-documenter/src/markdown/MarkdownEmitter.ts#L141-L147

`DocNodeTransforms.trimSpacesInParagraph()` returns a paragraph with one node `@docCategory` (`DocInlineTag`) whose rendering is an empty string.  Because we are rendering in linear traversal, it seems that it's not easy to determine whether the paragraph will be empty without actually calling `this.writeNodes()`.

This same problem probably exists in the more common `!docNodeSiblings` code path, but the empty paragraphs don't cause trouble there because they get folded together by `writer.ensureNewLine()`.  (Which technically is incorrect as well -- if we're rendering accurately than an empty paragraph should get emitted as something like `&#32;` to force the Markdown engine to make an empty paragraph.)

The `<p></p>`  tags do seem to make the rendered output look worse, so we should fix them.  Would you have any time to investigate this part?  I'd also be okay with handling it as a separate PR.

Reply:
GitHub's handling of `<p>` is a rather weird compared to [CommonMark](https://spec.commonmark.org/dingus/?text=a%0A%0A%3Cp%3Eb%3C%2Fp%3E%0A%0A%3Cp%3Ec%3C%2Fp%3E%3Cp%3Ed%3C%2Fp%3E%0A%0A):

Input:
```markdown
| a | b |
|--|--|
|c|<p>d</p>|
|e|<p>f</p><p>g</p>|
```

Output:
| a | b |
|--|--|
|c|<p>d</p>|
|e|<p>f</p><p>g</p>|

(Life would be soooo much easier if we rendered to a well-defined format like HTML or JSX instead of Markdown, but `api-documenter` -> `.md` is an immensely useful target heheh.)

Reply:
Yes! I certainly have time to investigate the `<p></p>` bug. I'd want to do it in a separate PR, if that's okay with you. Should this PR be blocked by that bug then? It looks like this PR will introduce a subtle rendering regression due to the `<p></p>` stuff, but maybe it's minor enough to live with it the short amount of time between when this PR goes in and the subsequent fix PR goes in.

Reply:
If you're not in a hurry to get this feature released for API Documenter, then it might be better to avoid introducing a regression.

Another option would be to merge+publish your changes here for `api-extractor` and `api-extractor-model`, but split off the `api-documenter` part of this work into a separate PR where we also solve the `<p></p>` problem.  It means this PR would go in without any test coverage, but I'm okay with that, given that we /did/ test it.

Reply:
> Another option would be to merge+publish your changes here for `api-extractor` and `api-extractor-model`, but split off the `api-documenter` part of this work into a separate PR where we also solve the `<p></p>` problem. It means this PR would go in without any test coverage, but I'm okay with that, given that we /did/ test it.

Let's do that! I'll split off the `api-documenter` piece.

Reply:
Okay, it should be split off now. LMK if you see any issues!